src/app/utils/sample-url-generation.ts

```diff
@@ -82,6 +82,33 @@ export function removeExtraSlashesFromUrl(url: string): string {
 
 export function hasWhiteSpace(url: string): boolean {
   const parts = url.split('?');
-  const whitespaceChars = [' ', '\t', '\n', '%20'];
-  return whitespaceChars.some((char) => parts[0].includes(char));
+  const whitespaceChars = [' ', '\t', '\n', '%20', '\r'];
+  const allParts = url.split(/(%20|\s)/);
+  return getWhiteSpace(parts, allParts, whitespaceChars);
+}
+
+export function getWhiteSpace(parts: string[], allParts: string[], whitespaceChars: string[]): boolean {
```

Comment:
*Nitpick*: The function is checking the presence of whitespaces. The name `getWhiteSpace` suggests we are going to do something with the whitespaces. This can be renamed to something clearer or of a boolean nature. Like `whiteSpaceExists`